### ISSUE TITLE
Discovery failing for invalid hostnames even when discovery_by_ip enabled

### ISSUE THREAD

**Issue by towster** [librenms/librenms#5525]
DO NOT DELETE THIS INFORMATION.

> Please read this information carefully.

GitHub issues is for feature requests or bugs, please do not post issues asking for help or how to do X, Y or Z. 
You can use our irc channel ##librenms on freenode to ask questions or our [community site](https://community.librenms.org).

Please confirm each of the sections below by putting an `x` in the box like `[x]`.

- [x] Is your install up to date? [Updating your install](http://docs.librenms.org/General/Updating/)
      Please do not submit an issue if your install is not up to date within the last 24 hours or on a stable monthly release.
- [x] Please include all of the information between the `====================================` section of `./validate.php` which you can run from the cli.
- [x] Unless your issue is for a WebUI fix or feature then please provide ALL info asked for [here](http://docs.librenms.org/Support/FAQ/#faq20).
- [x] Please provide as much detail as possible.
- [x] Please do NOT post more than 10 lines of debug information here, use a pastebin service or GitHub Gists.

==========================================================
Component | Version
--------- | -------
LibreNMS  | 81bde0b8da6c798245a7916ac6e2096a9c4ab1cc
DB Schema | 156
PHP       | 7.0.14
MySQL     | 5.5.52-MariaDB
RRDTool   | 1.4.8
SNMP      | NET-SNMP 5.7.2
==========================================================

Lots of info on the issue is posted here:
https://community.librenms.org/t/discovery-sees-devices-but-are-not-added-to-lnms/438/9

TL;DR; The two devices that are missing are:

        [cdpCacheDeviceId] => DC-NEXUS-5K-H7-2(SSI144007HH)
        [cdpCacheDeviceId] => DC-NEXUS-5K-H7-1(SSI144007GJ)

They are failing to pass this condition in discovery-protocols.inc.php:

            if (is_valid_hostname($cdp['cdpCacheDeviceId'])) {

I am thinking it should be changed to something like:

                if (is_valid_hostname($cdp['cdpCacheDeviceId']) || ($config['discovery_by_ip'] == true)) {

I have tested that on my install and will submit patch for this..
